File: katas/2021-06-08 Change Return/solutions/ChangeReturnKata_aScharfe/ChangeReturnKata/BusinessObject/Change.cs
namespace ChangeReturnKata.BusinessObject
{
    public class Change
    {
        public int HundredEuro { get; }
        public int FiftyEuro { get; }
        public int TwentyEuro { get; }
        public int TenEuro { get; }
        public int FiveEuro { get; }

        public int FiftyCent { get; }
        public int TwentyCent { get; }
        public int TenCent { get; }
        public int FiveCent { get; }
        public int TwoCent { get; }
        public int OneCent { get; }

        /// <summary>
        /// Constructor for Change. Calculates the change.
        /// </summary>
        /// <param name="changeReturn">Decimal value.</param>
        public Change(decimal changeReturn)
        {
            HundredEuro = (int) (changeReturn / 100);
            changeReturn %= 100;
            FiftyEuro = (int) (changeReturn / 50);
            changeReturn %= 50;
            TwentyEuro = (int) (changeReturn / 20);
            changeReturn %= 20;
            TenEuro = (int)(changeReturn / 10);
            changeReturn %= 10;
            FiveEuro = (int) (changeReturn / 5);
            changeReturn %= 5;
            FiftyCent = (int) (changeReturn / .50m);
            changeReturn %= .50m;
            TwentyCent = (int) (changeReturn / .20m);
            changeReturn %= .20m;
            TenCent = (int) (changeReturn / .10m);
            changeReturn %= .10m;
            FiveCent = (int) (changeReturn / .05m);
            changeReturn %= .05m;
            TwoCent = (int) (changeReturn / .02m);
            changeReturn %= .02m;
            OneCent = (int) (changeReturn / .01m);
        }
    }
}

File: aScharfe/ChangeReturnKata_Python_aScharfe/ChangeReturnKata_Python/Calculator/Change.py
from decimal import Decimal

class Change(object):
    def __init__(self,changeReturn):
        self.hundredEuro = int(changeReturn / 100)
        changeReturn %= 100        
        
        self.fiftyEuro = int(changeReturn / 50)
        changeReturn %= 50
        
        self.twentyEuro = int(changeReturn / 20)
        changeReturn %= 20
        
        self.tenEuro = int(changeReturn / 10)
        changeReturn %= 10
        
        self.fiveEuro = int(changeReturn / 5)
        changeReturn %= 5        
        
        self.fiftyCent = int(changeReturn / Decimal('0.5'))        
        changeReturn %= Decimal('0.5')

        self.twentyCent = int(changeReturn / Decimal('0.2'))        
        changeReturn %= Decimal('0.2')

        self.tenCent = int(changeReturn / Decimal('0.1'))        
        changeReturn %= Decimal('0.1')

        self.fiveCent = int(changeReturn / Decimal('0.05'))        
        changeReturn %= Decimal('0.05')

        self.twoCent = int(changeReturn / Decimal('0.02'))        
        changeReturn %= Decimal('0.02')

        self.oneCent = int(changeReturn / Decimal('0.01'))

   


